cjworkbench/i18n/catalogs/extract.py

```diff
@@ -116,6 +178,11 @@ def extract_python(fileobj, keywords, comment_tags, options):
     :param comment_tags: a list of translator tags to search for and include
                          in the results
     :param options: a dictionary of additional options (optional)
+    :param process: a callable that accepts funcname, messages, translator_comments, message_lineno
```

Comment:
Why a callback? Wouldn't it be more Pythonic for the caller to use a simple for-loop:

```
for message_lineno, funcname, messages, translator_comments in _parse_and_extract_python(...):
    if ...:
        yield funcname, messages, translator_comments, message_lineno
```

Also, with this approach, we could guarantee `extract_python()` won't return `funcname=None`.

Reply:
The `funcname` returned by `extract_python` may need to be `None`. `pybabel` expects only `gettext`-kind function names.

Reply:
Ohhhhhh, gotcha.